### ISSUE TITLE
No Content Left Behind

### ISSUE THREAD

**Issue by mehcode** [atom/atom#6455]
From https://github.com/atom/atom/pull/6380#issuecomment-94878888
> I think it's critical. It's not about atom crashing or not crashing. It's about a myriad of other things crashing on a computer. It's about being able to not kill a junior because he or she accidentally hit the reset button on your computer. Sublime gives you the "don't worry about it" guarantee. It's a state of mind and affects your workflow. I was used to having many unnamed buffers open with random scratch data. I didn't need to save it anywhere but sublime kept them around.

-----

From @maxbrunsfeld on https://github.com/atom/atom/pull/6380#issuecomment-94881588
> I think this use case is covered though, because Atom should quit normally during a system restart (at least on OSX and Linux).

Not when I hard reset my computer with a physical reset button which is quite accessible and happens to be right next to the USB ports.

-----

Open atom, type some text, and hit `Ctrl+Q`

![screenshot from 2015-04-21 10 40 49](https://cloud.githubusercontent.com/assets/753919/7258673/e6d837c8-e812-11e4-8232-c4f8d736fb5c.png)

I should not be able to lose work at all, at any time, for any reason. In sublime I can do the above and then immediately re-open and all is well.

-----

To summarize, we need a few things:
 - Save unnamed buffers when atom is closed normally
 - Porting the "hot exit" feature from sublime. When atom is closed, it is immediately closed without any popup and can be re-opened in the exact state it was left
 - On some configurable interval we save the state to allow the application to pick back up in case of fire

**Comment by Pholey** [comment]
:+1:

**Comment by mnquintana** [comment]
Duplicate of #942?

**Comment by mehcode** [comment]
Perhaps but I was informed on https://github.com/atom/atom/pull/6380#issuecomment-94881588 to open an issue for it

**Comment by eddiemonge** [comment]
I partially disagree on this change. While its nice that atom remembers my tabs and projects (usually), I would not want it to remember code that I intentionally did not save.

**Comment by mehcode** [comment]
@eddiemonge But what if it remembers everything? From code you didn't save to your undo buffer to undo that code you didn't save? It's not auto-saving; it's just remembering your buffers. 

Of course this behavior should be opt-out.

**Comment by niedzielski** [comment]
This is the only feature keeping me from switching from Sublime. It was very surprising to me when my computer did crash and I did lose a good bit of work, something that's not possible in Sublime or Vim. It only takes one loss for this to become a must have feature, at least for me.

**Comment by maxbrunsfeld** [comment]
> It does look like #942 is what I am talking about here.

@mehcode Ok, then we should close this in favor of that earlier issue. My apologies for telling you to open another one; I didn't fully understand the feature from Sublime that you were referring to.
